client/components/tinymce/plugins/insert-menu/plugin.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import tinymce from 'tinymce/tinymce';
+import { renderToString } from 'react-dom/server';
+
+import Gridicon from 'components/gridicon';
+import i18n from 'lib/mixins/i18n';
+
+import menuItems from './menu-items';
+
+const initialize = editor => {
+	menuItems.forEach( item =>
+		editor.addMenuItem( item.name, {
+			classes: 'wpcom-insert-menu__menu-item',
+			cmd: item.cmd,
+			onPostRender() {
+				this.innerHtml( renderToString( item.item ) );
+			}
+		} )
+	);
+
+	editor.addButton( 'wpcom_insert_menu', {
+		type: 'splitbutton',
+		title: i18n.translate( 'Insert content' ),
+		classes: 'btn wpcom-insert-menu insert-menu',
+		cmd: menuItems[0].cmd,
+		menu: menuItems.map( ( { name } ) => editor.menuItems[ name ] ),
+		onPostRender() {
+			ReactDOM.render(
+				<Gridicon icon={ menuItems[0].icon } />,
```

Comment:
I'm not sure we really need to be pulling this from `menuItems`, especially when the icon we're pulling is not even the same as used in the `<Gridicon />` the menu item renders (`add-image` vs. `image-multiple`). I think we'd be safe in hard-coding this to `add-image` and dropping the `icon` property from `menuItem` entries. What do you think?